java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -703,6 +690,78 @@ protected Map<String, Object> handleLcsIdxResponse(Map<String, Object> response)
         return response;
     }
 
+    /**
+     * @param response A Protobuf response
+     * @return A map of a map of <code>String[][]</code>
+     */
+    protected Map<String, Map<String, String[][]>> handleXReadResponse(Response response)
+            throws RedisException {
+        Map<String, Object> mapResponse = handleMapOrNullResponse(response);
+        if (mapResponse == null) {
+            return null;
+        }
+        return mapResponse.entrySet().stream()
+                .collect(
+                        Collectors.toMap(
+                                Map.Entry::getKey,
+                                e -> castMapOf2DArray((Map<String, Object[][]>) e.getValue(), String.class)));
+    }
+
+    /** Converts array types in the response from `XINFO STREAM` to be more user-friendly. */
+    @SuppressWarnings("unchecked")
+    protected Map<String, Object> handleXInfoStreamResponse(Map<String, Object> map) {
+        // convert entries from `Object[][]` to `String[][]` inside a map
+        for (var keyName : List.of("entries", "first-entry", "last-entry")) {
+            if (!map.containsKey(keyName)) continue;
+            var entries = (Map<String, Object[][]>) map.get(keyName);
```

Comment:
@acarbonetto @Yury-Fridlyand Currently we are returning an array of Objects where the first element is the key and second is an array of Objects. It seems like we should convert to Map<String, Object>. Should we look in to this or should this remain this way similar to we did in [GeoSearch](https://github.com/aws/glide-for-redis/blob/4ab00ddc05f109c79bd36077ad080e81f54b59fc/java/client/src/main/java/glide/api/commands/GeospatialIndicesBaseCommands.java#L476) cc @GumpacG 
![Screenshot 2024-07-05 at 11 48 39 AM](https://github.com/aws/glide-for-redis/assets/111323543/86d2edab-34ad-4692-8e40-be8ed01d9a29)

Reply:
Here we are casting the values associated with each keyName to a`(Map<String, Object[][]>)`, but the values associated with each of those keys are actually arrays (see [here](https://github.com/aws/glide-for-redis/blob/main/python/python/glide/async_commands/core.py#L3581) and [here](https://github.com/aws/glide-for-redis/blob/main/python/python/glide/async_commands/core.py#L3627)). I think we originally planned for thes "first-entry", "last-entry", and "entries" values to be maps, but the value conversion was really nasty and after talking with AWS we decided to leave them as arrays.